src/dist/http/http_server.cpp

```diff
@@ -127,23 +127,29 @@ void http_server::add_service(http_service *service)
         unresolved_query = decoded_unresolved_query.get_value();
     }
 
+    // remove tailing '\0'
+    if (!unresolved_path.empty() && *unresolved_path.crbegin() == '\0') {
+        unresolved_path.pop_back();
+    }
```

Comment:
why remove? is it bug?

Reply:
because we don't want the url path includes '\0'